subprojects/core/src/main/java/org/gradle/initialization/DefaultSettings.java

```diff
@@ -101,11 +105,21 @@ public DefaultSettings(
         this.settingsScript = settingsScript;
         this.startParameter = startParameter;
         this.services = serviceRegistryFactory.createFor(this);
-        this.rootProjectDescriptor = createProjectDescriptor(null, settingsDir.getName(), settingsDir);
+        this.rootProjectDescriptor = createProjectDescriptor(null, getProjectName(settingsDir), settingsDir);
         this.dependencyResolutionManagement = services.get(DependencyResolutionManagementInternal.class);
         this.toolchainManagement = services.get(ToolchainManagementInternal.class);
     }
 
+    private static String getProjectName(File settingsDir) {
+        if(contains(File.listRoots(), settingsDir)) {
```

Comment:
```suggestion
        if (contains(File.listRoots(), settingsDir)) {
```

💄  In general, I think we use a space in front of the `(` after an `if`.

Reply:
That happens because I'm currently a bit afraid of using the auto formatter ;)